### PR TITLE
Update copyright headers across the codebase

### PR DESCRIPTION
**Pull request by meta-paul**
This PR comprises a script to auto-update copyright headers `scripts/update_copyright_headers.py`, its output, and a few other manual changes.

### CODE REVIEW

File: scripts/update_copyright_headers.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        lines = f.readlines()
+
+    if len(lines) < 2:
+        raise UnsupportedFile("File has fewer than one line")
+
+    # Check copyright presence at the top of the file
+    anchor_line_number = -2
+    for i, line in enumerate(lines[:EXAMINED_LINES]):
+        text = line[:50].lower()
+        if "copyright" in text:
```
Comment (mojtaba-komeili) [inline scripts/update_copyright_headers.py:91]: Why do we need this branch, could we merge it with the next one?

File: scripts/update_copyright_headers.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+            anchor_line_number = -1
+        if anchor_line_number == -1 and ("meta" in text or "facebook" in text):
```
Comment (mojtaba-komeili) [inline scripts/update_copyright_headers.py]: Could it be `"copyright" in text` instead  of `anchor_line_number == -1` and removing the above line?

**Review by mojtaba-komeili — APPROVED** [review]
The script makes sense and the outcome is what we expected. LGTM.

Feel free to ignore my other comments. It made more sense after seeing the rest of the code.

**Comment by mojtaba-komeili** [comment]
Is that possible to fix the tests, maybe in a follow up PR?

File: scripts/update_copyright_headers.py (hunk 2 above)
Comment (meta-paul) [reply scripts/update_copyright_headers.py]: Agree, this logic could be written better (I abused integer line counter by using it as a status variable). Will push an update.

File: scripts/update_copyright_headers.py (hunk 1 above)
Comment (meta-paul) [reply scripts/update_copyright_headers.py:91]: Updating copyright is a different task from creating review app, or other tasks within release v1.2, so I've decided to create a separate branch. In addition, it has a large diff, and resolving potential conflicts is easier from a separate branch.

**Comment by meta-paul** [comment]
Unittests are broken here (not because copyright script broke the code, but rather it's been a long-standing issue). They will be fixed in a separate branch, and it's not an easy task.

**Review by meganung — APPROVED** [review]
offline discussion, lgtm !

**Pull request merged**